beacon_node/beacon_chain/src/data_availability_checker/overflow_lru_cache.rs

```diff
@@ -304,26 +284,25 @@ impl<E: EthSpec> PendingComponents<E> {
             return Err(AvailabilityCheckError::Unexpected);
         };
 
-        let (blobs, data_columns) = match block_import_requirement {
-            BlockImportRequirement::AllBlobs => {
-                let num_blobs_expected = diet_executed_block.num_blobs_expected();
-                let Some(verified_blobs) = verified_blobs
-                    .into_iter()
-                    .map(|b| b.map(|b| b.to_blob()))
-                    .take(num_blobs_expected)
-                    .collect::<Option<Vec<_>>>()
-                else {
-                    return Err(AvailabilityCheckError::Unexpected);
-                };
-                (Some(VariableList::new(verified_blobs)?), None)
-            }
-            BlockImportRequirement::ColumnSampling(_) => {
-                let verified_data_columns = verified_data_columns
-                    .into_iter()
-                    .map(|d| d.into_inner())
-                    .collect();
-                (None, Some(verified_data_columns))
-            }
+        let is_peer_das_enabled = spec.is_peer_das_enabled_for_epoch(diet_executed_block.epoch());
+        let (blobs, data_columns) = if is_peer_das_enabled {
+            let data_columns = verified_data_columns
+                .into_iter()
+                .map(|d| d.into_inner())
+                .collect::<Vec<_>>();
+            (None, Some(data_columns))
+        } else {
+            let num_blobs_expected = diet_executed_block.num_blobs_expected();
+            let Some(verified_blobs) = verified_blobs
+                .into_iter()
+                .map(|b| b.map(|b| b.to_blob()))
+                .take(num_blobs_expected)
```

Comment:
This is an existing check from Deneb to make sure we have exactly the same number of blobs as the kzg commitments, otherwise it returns `AvailabilityCheckError::Unexpected`

Reply:
Beyond the scope of this PR